brave/brave-core
# Get rid of IdentityManager call in ProfileSyncService

Resolves https://github.com/brave/brave-browser/issues/12984

The place in code where it happens is 
and here is a place in code where it happened
https://source.chromium.org/chromium/chromium/src/+/master:components/sync/driver/profile_sync_service.cc;l=962-970;drc=d81c5852498699fe3cd812e78d31c77c28e29281;bpv=1;bpt=1?q=ProfileSyncService::OnSyncCycleCompleted&ss=chromium%2Fchromium%2Fsrc

But we trying as much as possible to avoid patches. So in this PR we introduce own `BraveIdentityManagerFactory` which creates  `BraveIdentityManager` with stubbed method of `GetAccountsInCookieJar()` and `BravePrimaryAccountMutatorImpl` with stubbed method `ClearPrimaryAccount`

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.

## Test Plan:

1. Open tab brave://net-export/
2. Start network logging with high available option (` Include raw bytes (will include cookies and credentials)`)
3. Enable sync, ensure you can see the device
4. Stop network logging
5. Analyze the log (suggested https://netlog-viewer.appspot.com/#import) 
6. Ensure there is no activity to `accounts.google.com`
